GHE_main.cpp

```diff
@@ -0,0 +1,237 @@
+    vector<double> g_func = get <1> (test_data);
+    vector<double> q_load = get <2>(test_data);
+    
+    // from length of q_load, build 0 - 8760 integer vector called q_time.
+    vector<int>::size_type length = q_load.size();
+    vector<int> q_time(length);
+    iota(begin(q_time),end(q_time),1);
+    get<1>(g_data) = length ;
+    // for each value of q_time calculate ln(t/ts)
+    int i = 0;
+    vector <double> q_lntts;
+    for (int t : q_time){
+        q_lntts[i] = log (t/ts);
+        i++;}
+    // Create a matching condition that replaces each lntts value from q_lnnts to the closest value from lnnts
+    int j = 0;
+    vector <double> q_lntts_new;
+    for (double q_t : q_lntts){
+        for ( double time : lntts){
+            if (abs(q_t - time) < 0.2){q_lntts_new [j] = time ;}
+        }
+        j++;}
+    // Match the values (which will now look like steps) of q_lnnts to the corresponding g values
+    i = 0;
+    j = 0;
+    for(double q_t_new : q_lntts_new){
+        if (q_t_new == lntts [j]){get<0>(g_data)[i] = g_func [j];}
+        else {j++;}
+        i++;}
+    // Create a vector, g_data, with these values that is the same length as q_load and return g_data
+    if (get<0>(g_data).size() == q_load.size()){return g_data;}
+    else{puts("q_load size does not equal g_data size");
+              exit (EXIT_FAILURE);}}
+
+// eqn 1.11
+double summation(double n, vector<double> q_data, vector<double> g_data)
+{
+    double total = 0;
+    int i = 0;
+    while ( i < n ){
+        qi = q_data[i];
+        qi1 = q_data[i-1];
+        // eqn 1.1
+        // need to ensure variables passed to g are cosistent with what is required
+        total = total + (qi - qi1)* g_data[i];
+        ++i;}
+    return total;
+}
+
+/*---------------
+Main function
+----------------*/
+output_data simple_GHE(double Tin, double mdot)
+{
+//    //Debugging print functions:
+//    int j = 0;
+//    while (j < 5){
+//        printf("q at import for index %d is %f\n",j,q_array[j]);
+//        j = j + 1;
+//    }
+//    j = 0;
+//    while (j < 5){
+//        printf("g at import for index %d is %f\n",j,g_data[j]);
+//        j = j + 1;
+//    }
+    
+    // loading JSON data. In development
+    json_data input_data = read_json(); //this will be a function that reads and returns the data from the JSON file. See sndbx for variable type
+    vector<double> q_data = get<2>(input_data.indexed_data);
+    double cp = input_data.cp;
+    double H = input_data.H;
+    double Rb = input_data.Rb;
+    double ks = input_data.ks;
+    double Ts = input_data.Ts;
+    
+    //loading g_data
+    tuple<vector<double>, vector<int>::size_type> g_data = g_expander(input_data.indexed_data, ts);
+    
+    //Defining variables for use in fucntion
+    int n = get<1>(g_data);
+    qn1 = q_data[n];
+    n = n+1;
+    ts = pow(H,2)/(9*alphas);
+    alphas = 1*pow(10,-6);
+    
+
+    
+    //Debugging print functions:
+    printf("n is %d\n",n);
```

Comment:
When you use `#include <iostream>`, change this to: `std::cout << "n is " << n << std::endl;`.